src/auth/auth.service.ts

```diff
@@ -1,261 +1,132 @@
-  public async registerSellerCompany(
-    dto: SellerCompanyRegisterDto,
-  ): Promise<void> {
-    const { email } = dto;
-
-    const user = await this.userService.findOne({ email, emailVerified: true });
-
-    if (user) {
-      throw new EmailAlreadyExistsException(email);
-    }
-
-    await this.sellerService.createCompany(dto);
-
-    await this.sendVerificationMessage({ email, role: UserRole.SellerCompany });
-  }
-
-  async login(dto: UserLoginDto): Promise<
-    (BuyerEntity | IndividualEntity | CompanyEntity) & {
-      token: string;
-      role: string;
-    }
-  > {
-    const { email, password } = dto;
-
-    const user = await this.userService.findOne({ email, emailVerified: true });
-
-    if (!user) {
-      throw new NotFoundException('User not found');
-    }
-
-    const isPasswordCorrect = await compare(password, user.password);
-
-    if (!isPasswordCorrect) {
-      throw new BadRequestException('Invalid password provided');
-    }
-
-    const { role, id } = user;
-
-    const userData = await this.findUserByRole(role, { id });
-
-    if (!userData) {
-      throw new UnprocessableEntityException(
-        `User with role '${role}' and id '${id}' not found`,
+    if (buyerByEmail || buyerByPhoneNumber) {
+      throw new HttpException(
+        'Email or Phone is already in use ',
+        HttpStatus.UNPROCESSABLE_ENTITY,
       );
     }
-
-    return {
-      ...userData,
-      token: this.generateJwt({ id: userData.id, email, role }),
-      role,
-    } as (BuyerEntity | IndividualEntity | CompanyEntity) & {
-      token: string;
-      role: string;
-    };
+    const savedUser = await this.userRepository.save(user);
+    savedUser.buyer = buyer;
+    await this.userRepository.save(savedUser);
+    const savedBuyer = await this.buyerRepository.save(buyer)
+    return savedBuyer;
   }
 
-  private generateJwt(user: JwtPayload): string {
-    const jwtSecret = this.configService.get<string>('JWT_SECRET');
 
-    return jwt.sign(
-      {
-        id: user.id,
-        email: user.email,
-      },
-      jwtSecret,
-    );
-  }
-
-  public async sendVerificationMessage(
-    data: VerificationTokenPayload,
-  ): Promise<void> {
-    const secret = this.configService.get<string>('JWT_VERIFICATION_SECRET');
-    const expiresIn = this.configService.get<string>(
-      'JWT_VERIFICATION_EXPIRATION',
-    );
-    const token = jwt.sign(data, secret, { expiresIn });
-
-    await this.mailService.sendVerificationLetter({
-      to: data.email,
-      data: { token },
+    async createSeller(createSellerDto: CreateSellerDto): Promise<SellerEntity> {
+      const user = new UserEntity();
+      console.log ('email', user.email)
+    user.email = createSellerDto.email;
+      user.password = createSellerDto.password;
+      user.role = Role.Seller;
+    const seller = new SellerEntity();
+    seller.companyName = createSellerDto.companyName;
+    Object.assign(seller, createSellerDto);
+      console.log('newSeller', seller)
```

Comment:
remove console.log and fix the formatting of the next few lines

Reply:
✅